scripts/modules/events/backward-compatibility.js

```
@@ -0,0 +1,430 @@
+ * @param {Object} payload - Modern payload
+ * @param {string} targetVersion - Target legacy version
+ * @returns {Object} Legacy-compatible payload
+ */
+export function createLegacyCompatiblePayload(
+	payload,
+	targetVersion = LEGACY_VERSIONS['0.1.0']
+) {
+	switch (targetVersion) {
+		case LEGACY_VERSIONS['0.1.0']:
+			// Convert back to wrapped format for pre-schema integrations
+			const { version, eventId, timestamp, context, ...eventData } = payload;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)